### PR TITLE
[ci skip] Update contributing guidelines

### PR DESCRIPTION
**Pull request by kennytv**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -99,12 +97,13 @@ of how to use git is required. A basic tutorial can be found here:
 Assuming you have already forked the repository:
 
 1. Clone your fork to your local machine;
-1. Type `./paper patch` in a terminal to apply the changes from upstream;
-1. cd into `Paper-Server` for server changes, and `Paper-API` for API changes.  
-You can also run `./paper server` or `./paper api` for these same directories
+2. Type `./gradlew applyPatches` in a terminal to apply the changes from upstream.
+On Windows, leave out the `./` at the beginning for all `gradlew` commands;
+3. cd into `Paper-Server` for server changes, and `Paper-API` for API changes.  
+<!--You can also run `./paper server` or `./paper api` for these same directories
```
Comment (MiniDigger) [inline CONTRIBUTING.md:106]: this is commented out, do we wanna restore that behavior? then we should have an issue. if not, we should just nuke this reference.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -449,21 +451,21 @@ patching process.
 progress will be lost if you do not;
 1. Identify the name(s) of the file(s) you want to import.
    - A complete list of all possible file names can be found at
-   `./work/Minecraft/$MCVER/spigot/net/minecraft/server`. You might find
+   `./Paper-Server/.gradle/caches/paperweight/mc-dev-sources/net/minecraft/`. You might find
```
Comment (MiniDigger) [inline CONTRIBUTING.md:464]: I _think_ that folder is only generated if you execute some task, right?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -449,21 +451,21 @@ patching process.
...
-> undo your changes to `importmcdev.sh`.
+> undo your changes to `dev-imports.txt`.
 
 Any file modified in a patch file gets automatically imported, so you only need
 this temporarily to import it to create the first patch.
 
-To undo your changes to the file, type `git checkout scripts/importmcdev.sh`.
+To undo your changes to the file, type `git checkout build-data/dev-imports.txt`.
 
 ### Where can I learn how to name method/field?
```
Comment (MiniDigger) [inline CONTRIBUTING.md]: is this still true?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -514,13 +516,7 @@ everything like usual.
 
 To install the API to your local maven repository, do the following:
 
-- Enter the API directory by running `./paper api`;
-- Run `mvn install`.
-   - If you are working on a patch without much care for whether the tests
```
Comment (MiniDigger) [inline CONTRIBUTING.md:519]: why did you remove the info about the tests? is there a gradle property to skip tests?

**Review by MiniDigger — COMMENTED** [review]
thank you for taking the lead on fixing this, somebody had to do it :D

generally looks good to me, just minor comments

File: CONTRIBUTING.md (hunk 2 above)
Comment (kennytv) [reply CONTRIBUTING.md:464]: it's autogenerated on applyPatches since https://github.com/PaperMC/paperweight/pull/57 (and https://github.com/PaperMC/paperweight/pull/58)

File: CONTRIBUTING.md (hunk 3 above)
Comment (kennytv) [reply CONTRIBUTING.md]: Yessir, used `build-data/dev-imports.txt` today to verify

File: CONTRIBUTING.md (hunk 4 above)
Comment (kennytv) [reply CONTRIBUTING.md:519]: Apparently that doesn't actually run tests in the first place now (otherwise it would be `-x test` iirc)

File: CONTRIBUTING.md (hunk 3 above)
Comment (MiniDigger) [reply CONTRIBUTING.md]: I meant the stuff below, about naming stuff using yarn names, using mojang only as a reference but never verbatim. (github doesn't let me comment on stuff outside of the scope of the PR)

File: CONTRIBUTING.md (hunk 1 above)
Comment (kennytv) [reply CONTRIBUTING.md:106]: There's an open pr at least https://github.com/PaperMC/Paper/pull/6009

File: CONTRIBUTING.md (hunk 3 above)
Comment (kennytv) [reply CONTRIBUTING.md]: Oh right true, that section can probably be removed entirely since the only names we have to come up with are the ones that aren't included in Yarn or Mojmaps (or Spigot dummery), which is covered in the obfuscation section

File: CONTRIBUTING.md (hunk 1 above)
Comment (MiniDigger) [reply CONTRIBUTING.md:106]: thats fine then, just gotta remember to uncomment again :D

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -352,31 +349,18 @@ index a92bf8967..d0ab87d0f 100644
...
-```
-
-Or it may be as complex as forwarding an entire method so that it can be
-overridden later:
+While rarely needed, obfuscation helpers are sometimes useful when it comes
+to unmapped local variables, or poorly named method parameters. In an effort
+to make future updates easier on ourselves, Paper tries to use obfuscation
+helpers wherever it makes sense. The purpose of these helpers is to make the
+code more readable and maintainable. These helpers should be made as easy to
+inline as possible by the JVM whenever possible.
```
Comment (MiniDigger) [inline CONTRIBUTING.md]: As easy as possible whenever possible feels kinda doubled, but whatever

**Review by MiniDigger — APPROVED** [review]
Lgtm

File: CONTRIBUTING.md (hunk 5 above)
Comment (jpenilla) [reply CONTRIBUTING.md]: blame Z https://github.com/PaperMC/Paper/commit/4da4bd812a17fe34b35605f9bfaa46aed5216034#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R115
